Microsoft.Research/Contracts/System.Core.Contracts/System.Linq.Expressions.DynamicExpression.cs

```diff
@@ -49,34 +51,39 @@ protected internal override Expression Accept (ExpressionVisitor visitor)
       return default(Expression);
     }
 
+    [Pure]
     public DynamicExpression Update (IEnumerable<Expression> arguments)
     {
+      Contract.Ensures(Contract.Result<DynamicExpression>() != null);
```

Comment:
> There are many methods in Expression that don't require non-null on `IEnumerable<>` and params arrays for that reason. Should those be changed as well?

I'd like to discuss this after I finish reviewing it, and we can make a decision on whether to leave them all as-is or change them all.